specification/hybridconnectivity/HybridConnectivity.Management/EndpointResource.tsp

```diff
@@ -0,0 +1,167 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.HybridConnectivity;
+/**
+ * The endpoint for the target resource.
+ */
+model EndpointResource
+  is Azure.ResourceManager.ExtensionResource<EndpointProperties> {
+  ...ResourceNameParameter<
+    Resource = EndpointResource,
+    KeyName = "endpointName",
+    SegmentName = "endpoints",
+    NamePattern = ""
+  >;
+}
+
+@armResourceOperations
+interface EndpointResources {
+  /**
+   * Gets the endpoint to the resource.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @operationId("Endpoints_Get")
+  get is ArmResourceRead<
+    EndpointResource,
+    Azure.ResourceManager.Foundations.ExtensionBaseParameters
+  >;
```

Comment:
@pshao25 

Do you know why the `Endponts` -> `EndpointResources` happens in convertion?

It appears Java SDK's break is caused by this pattern (same as the `ServiceConfigurations`)
https://github.com/Azure/azure-rest-api-specs/pull/32243#issuecomment-2603764968

Reply:
SDK can use `@clientName` to override this, so not a blocking issue for SDK.
https://github.com/Azure/azure-rest-api-specs/pull/32243/files